File: Back/ATVComplementarPW.Domain/Entities/Transport.cs
namespace ATVComplementarPW.Domain.Entities;

public class Transport : BaseEntity
{
    public int VehicleId { get; set; }
    public Vehicle Vehicle { get; set; }
    public int PassengerId { get; set; }
    public Passenger Passenger { get; set; }
    public DateTime DateHourTransport { get; set; }
    public double TransportKm { get; set; }
    public double Price { get; protected set; }
    protected Transport()
    {
        Price = 0.4 * TransportKm;
        Active = true;
    }
    public Transport(int vehicleId, int passengerId, DateTime dateHourTransport, double transportKm)
    {
        VehicleId = vehicleId;
        PassengerId = passengerId;
        DateHourTransport = dateHourTransport;
        TransportKm = transportKm;
        Price = transportKm * 0.4;
        Active = true;
    }
}

File: Front/src/app/Models/transport.ts
import { Passenger } from "./passenger";
import { Vehicle } from "./vehicle";

export interface Transport {
  id: number;
  vehiclePlate: string;
  passengerCpf: string;
  dateHourTransport: string;
  transportKm: number;
  passenger?: Passenger;
  vehicle?: Vehicle;
}
